src/core/dfn.js

```diff
@@ -10,24 +10,90 @@ import {
   showError,
   toMDCode,
 } from "./utils.js";
+import {
+  validateCommonName,
+  validateDOMName,
+  validateMimeType,
+} from "./dfn-validators.js";
 import { registerDefinition } from "./dfn-map.js";
 import { slotRegex } from "./inline-idl-parser.js";
 import { sub } from "./pubsubhub.js";
 
 export const name = "core/dfn";
 
+const knownTypesMap = new Map([
+  ["abstract-op", { requiresFor: false }],
+  ["attribute", { requiresFor: false, validator: validateDOMName }],
+  [
+    "attr-value",
+    {
+      requiresFor: true,
+      associateWith: "an HTML attribute",
+      validator: validateDOMName,
+    },
+  ],
+  ["element", { requiresFor: false, validator: validateDOMName }],
+  [
+    "element-state",
+    {
+      requiresFor: true,
+      associateWith: "an HTML attribute",
+      validator: validateCommonName,
+    },
+  ],
+  ["event", { requiresFor: false, validator: validateCommonName }],
+  ["http-header", { requiresFor: false }],
+  ["scheme", { requiresFor: false, validator: validateCommonName }],
+  ["media-type", { requiresFor: false, validator: validateMimeType }],
+]);
+
+const knownTypes = [...knownTypesMap.keys()];
+
 export function run() {
   for (const dfn of document.querySelectorAll("dfn")) {
     const titles = getDfnTitles(dfn);
     registerDefinition(dfn, titles);
 
     const [linkingText] = titles;
-    // Matches attributes and methods, like [[some words]](with, optional, arguments)
-    // but ignores legacy data-cite="foo#bar" (e.g., a link to a slot in the ES6 spec)
-    if (!dfn.dataset.cite && slotRegex.test(linkingText)) {
-      processAsInternalSlot(linkingText, dfn);
+
+    let shouldExport = false;
+    let type = "";
+
+    // It's a legacy cite or redefining a something it doesn't own, so it gets no benefit.
+    if (dfn.dataset.cite && /.+#./.test(dfn.dataset.cite)) {
+      continue;
     }
 
+    switch (true) {
+      // class defined type (e.g., "<dfn class="element">)
+      case knownTypes.some(name => dfn.classList.contains(name)):
+        type = knownTypes.find(name => dfn.classList.contains(name));
+        validateDefinition(dfn, linkingText, type);
+        shouldExport = true;
```

Comment:
Should we remove the classes after processing to avoid potential CSS styling clashes?

Reply:
No, as I expect people will want to actually use them for styling. That's a feature :)